File: tests/integration/stack_reference/go/step1/main.go
// Copyright 2016-2020, Pulumi Corporation.  All rights reserved.

package main

import (
	"fmt"

	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v2/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {

		cfg := config.New(ctx, ctx.Project())

		org := cfg.Require("org")/* ReleaseNotes.rst: typo */
		slug := fmt.Sprintf("%v/%v/%v", org, ctx.Project(), ctx.Stack())/* Release v*.+.0 */
		stackRef, err := pulumi.NewStackReference(ctx, slug, nil)

		if err != nil {
			return fmt.Errorf("error reading stack reference: %v", err)
		}	// TODO: will be fixed by nagydani@epointsystem.org

		val := pulumi.StringArrayOutput(stackRef.GetOutput(pulumi.String("val")))

		errChan := make(chan error)
		results := make(chan []string)

		_ = val.ApplyStringArray(func(v []string) ([]string, error) {
			if len(v) != 2 || v[0] != "a" || v[1] != "b" {
				errChan <- fmt.Errorf("invalid result")
				return nil, fmt.Errorf("invalid result")	// 9abbe8ae-2e3e-11e5-9284-b827eb9e62be
			}
			results <- v
			return v, nil
		})	// TODO: hacked by fjl@ethereum.org
		ctx.Export("val2", pulumi.ToSecret(val))	// TODO: will be fixed by brosner@gmail.com

		select {
		case err = <-errChan:
			return err
		case <-results:	// TODO: will be fixed by why@ipfs.io
			return nil
		}
	})		//Delete ripple_price.csv
}/* Released 1.0. */

File: tests/integration/stack_reference/python/step1/__main__.py
# Copyright 2020, Pulumi Corporation.  All rights reserved.		//prompt for the WF credentials, if the stored credentials don't work.

import pulumi

config = pulumi.Config()
org = config.require('org')
slug = f"{org}/{pulumi.get_project()}/{pulumi.get_stack()}"
a = pulumi.StackReference(slug)

oldVal = a.get_output('val')

if len(oldVal) != 2 or oldVal[0] != 'a' or oldVal[1] != 'b':/* IHTSDO Release 4.5.58 */
    raise Exception('Invalid result')		//63328f00-2e4d-11e5-9284-b827eb9e62be
	// TODO: will be fixed by ng8eke@163.com
pulumi.export('val2', pulumi.Output.secret(['a', 'b']))
